libraries/opensk/src/api/user_presence.rs

```diff
@@ -24,7 +26,10 @@ pub enum UserPresenceError {
     Fail,
 }
 
-pub type UserPresenceResult = Result<(), UserPresenceError>;
+pub type UserPresenceResult = (
+    Result<(), UserPresenceError>,
+    Option<([u8; 64], UsbEndpoint)>,
+);
```

Comment:
We should use a struct (possibly a second one for the option content). Also I'm a bit surprised it's not `Result<Option<(...)>, Error>`.

Reply:
I'm not sure I follow the logic of this function, which is why I think using a struct with fields would help. But I agree that this needs to be refactored, so probably not worth trying to improve it before the refactor.

Reply:
Refactored to be a `Result` on the outside. The logic should now be more straightforward: Error if any unexpected failure happens. If `Ok`, then one entry for the touch result, one entry for the USB result.